app/grandchallenge/algorithms/models.py

```diff
@@ -69,6 +69,76 @@
 JINJA_ENGINE = sandbox.ImmutableSandboxedEnvironment()
 
 
+class AlgorithmInterfaceManager(models.Manager):
+
+    def create(
+        self,
+        *,
+        inputs,
+        outputs,
+        **kwargs,
+    ):
+        obj = super().create(**kwargs)
+        obj.inputs.set(inputs)
+        obj.outputs.set(outputs)
```

Comment:
This is not idempotent right? This should implement what you have in the data migration I think:

```suggestion
        obj = get_existing_interface_for_inputs_and_outputs(
            model=AlgorithmInterface, inputs=inputs, outputs=outputs
        )
        if not obj:
            obj = super().create(**kwargs)
            obj.inputs.set(inputs)
            obj.outputs.set(outputs)
```

Reply:
I'm not sure I understand. This is what we do in the migration as well. Calling this method twice with the same inputs and outputs would only result in 1 interface because of the call to `get_existing_interface_for_inputs_and_outputs`, so it is idempotent. What am I missing?

Reply:
In the implementation that is on the branch right now doing:

```python
inputs = [...]
outputs = [...]
AlgorithmInterface.objects.create(inputs=inputs, outputs=outputs)
AlgorithmInterface.objects.create(inputs=inputs, outputs=outputs)
```

will create two `AlgorithmInterface`s with the same inputs and outputs, right? I think that the suggestion will mean that only one is created and re-used on subsequent calls.

Reply:
Oh wow, I read your suggestion as my code. I think I need another coffee. Yes, you are right, of course. Sorry!

Reply:
Haha, it's okay, roll on the weekend!

Reply:
I don't know if passing the model to `get_existing_interface_for_inputs_and_outputs` here is right as that hardcodes it. You might be able to do `self.model` instead.

Reply:
You don't actually need it because it defaults to `AlgorithmInterface`. I just added the option to pass the model so that we can use this function in the migration where we need to work with historical models. 

Or is setting a default in `get_existing_interface_for_inputs_and_outputs` not a good idea ?